mp_root_n.c

```diff
@@ -82,7 +85,7 @@ mp_err mp_root_u32(const mp_int *a, uint32_t b, mp_int *c)
 
       /* denominator */
       /* t3 = t1**(b-1) * b  */
-      if ((err = mp_mul_d(&t3, b, &t3)) != MP_OKAY)               goto LBL_ERR;
+      if ((err = mp_mul_d(&t3, (mp_digit)b, &t3)) != MP_OKAY)               goto LBL_ERR;
```

Comment:
Before I forget it again: with `MP_16BIT` an <del>`uint32_t`</del> `int` might be bigger  than an `mp_digit` and the value of `b` has only been checked to be smaller than or equal to `INT_MAX/2`.
This might be a problem at other places, too, please check.